crates/compiler/unify/src/unify.rs

```diff
@@ -2398,23 +2398,77 @@ fn close_uninhabited_extended_union(subs: &mut Subs, mut var: Variable) {
             }
             Structure(FlatType::TagUnion(_, ext))
             | Structure(FlatType::RecursiveTagUnion(_, _, ext)) => {
-                var = *ext;
+                var = ext.var();
             }
             _ => internal_error!("not a tag union"),
         }
     }
 }
 
+enum UnifySides<T, U> {
+    Left(T, U),
+    Right(U, T),
+}
+
+#[must_use]
+fn unify_tag_ext<M: MetaCollector>(
+    env: &mut Env,
+    pool: &mut Pool,
+    vars: UnifySides<TagExt, Variable>,
+    mode: Mode,
+) -> Outcome<M> {
+    let (ext, var, flip_for_unify) = match vars {
+        UnifySides::Left(ext, var) => (ext, var, false),
+        UnifySides::Right(var, ext) => (ext, var, true),
+    };
+    let legal_unification = match ext {
+        TagExt::Openness(_) => {
+            // Openness extensions can only unify with flex/rigids.
+            // Anything else is a change in the monomorphic size of the tag and not
+            // a reflection of the polymorphism of the tag, an error.
+            matches!(
+                env.subs.get_content_without_compacting(var),
+                FlexVar(..)
+                    | RigidVar(..)
+                    | FlexAbleVar(..)
+                    | RigidAbleVar(..)
+                    // errors propagate
+                    | Error,
+            )
+        }
+        TagExt::Any(_) => true,
+    }
+    // Tag unions are always extendable during specialization
+    || M::IS_LATE;
```

Comment:
strange formatting. could this `IS_LATE` not be the first argument to the `||`? looks like that would safe work in some cases

Reply:
yeah it could be. probably better to make it first anyway, less likely optimization will miss it